src/wallet/wallet.h

```diff
@@ -805,7 +805,7 @@ class CWallet final : public CCryptoKeyStore, private interfaces::Chain::Notific
     /**
      * populate vCoins with vector of available COutputs.
      */
-    void AvailableCoins(interfaces::Chain::Lock& locked_chain, std::vector<COutput>& vCoins, bool fOnlySafe=true, const CCoinControl *coinControl = nullptr, const CAmount& nMinimumAmount = 1, const CAmount& nMaximumAmount = MAX_MONEY, const CAmount& nMinimumSumAmount = MAX_MONEY, const uint64_t nMaximumCount = 0, const int nMinDepth = 0, const int nMaxDepth = 9999999) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
It seems fragile to rely on the fact that the fallback defaults are the same if no cc is passed or a default constructed cc is passed.

Reply:
hm, so these are the tradeoffs I see- 

**optional**: closer to current function signature, less duplicated code constructing default cc, don't have [this](https://github.com/bitcoin/bitcoin/pull/15906/files#r279143904) problem with callers 

**required**: expectations are more explicit from the POV of a caller, not a big of deal to make the change since this function isn't invoked frequently.

I don't feel familiar enough with the codebase / C++ to have much of an opinion here. Marco, how strongly do you hold the preference for required? Seems like luke-jr & empact both prefer keeping it optional.

Reply:
Oh sure, go for what luke and empact prefer. I consider it a stylistic preference.